### ISSUE TITLE
InterMineObjectFetcher.run() does not report a configuration error

### ISSUE THREAD

**Issue by JoeCarlson** [intermine/intermine#992]
When building a lucene index, if there is a misconfiguration in keyword_search.properties (such as a typo in index.references.Object = misspelledReference), then the run() method will terminate early in the finally{} block but will not report a run time error.

The ant task will finish and report a successful build, but the lucene index will not have all items in the index, depending on how many items were parsed before the exception was thrown.

There are 2 try blocks in the run() method. If an exception is thrown, iterateOverObjects will terminate and the exception will bubble up to the outermost block, print a warning message in the log but otherwise quietly return an incomplete list. I think the exception should either be handled within iterateOverObjects so that a more complete list is generated or a RunTime exception should be thrown to indicate a problem in the configuration

**Comment by julie-sullivan** [comment]
See #896

**Comment by julie-sullivan** [comment]
```
2015-06-16 15:14:03 WARN  org.intermine.api.lucene.InterMineObjectFetcher     - 
org.intermine.pathquery.PathException: Unable to resolve path 'Gene.monkeys': field 'monkeys' of class 

'org.intermine.model.bio.Gene' not found in model 'genomic'
        at org.intermine.pathquery.Path.initialise(Path.java:195)
        at org.intermine.pathquery.Path.<init>(Path.java:102)
        at org.intermine.api.lucene.InterMineObjectFetcher.getPathQuery(InterMineObjectFetcher.java:567)
        at org.intermine.api.lucene.InterMineObjectFetcher.handleObject(InterMineObjectFetcher.java:249)
        at org.intermine.api.lucene.InterMineObjectFetcher.iterateOverObjects(InterMineObjectFetcher.java:378)
        at org.intermine.api.lucene.InterMineObjectFetcher.run(InterMineObjectFetcher.java:168)
```

**Comment by julie-sullivan** [comment]
@justinccdev Can you take a look at this? I think Alex had a fix but we didn't merge it in, I haven't looked at it. See https://github.com/alexkalderimis/intermine/commit/e840012a4b1979178bf1d3828d3404d848382707.

**Comment by justinccdev** [comment]
The fix appears to work though it needed an addition to project.properties to get the ant build to work.  Pull request #1110

**Comment by justinccdev** [comment]
Okay, I see now that this fails checkstyle on too long methods and unused imports (I admit I forgot to run the tests locally).  I will need to go back and fix this.

**Comment by justinccdev** [comment]
Okay, I have now fixed this PR's checkstyle problems.  Checkstyle is still failing but that stems from pull request #1096

**Comment by julie-sullivan** [comment]
Fixed by #1128
